src/lib.rs

```diff
@@ -47,39 +48,45 @@ impl PrivateFS {
     })
   }
 
-  pub fn include(&self, path: impl AsRef<Path>) -> Result<(), Box<dyn Error>> {
+  pub fn include(&self, source_path: impl AsRef<Path>, destination_path: Option<&str>) -> Result<(), Box<dyn Error>> {
```

Comment:
An alternative here would be to add a generic parameter `S: AsRef<Path>` and tie both inputs to use `S`. 
We cannot have `destination_path: Option<impl AsRef<Path>>` because it causes an inference error when the path is omitted.

Reply:
@LukeMathWalker would it be possible to do this so they can be separate types or is the inference breaking here?

```rust
pub fn include(&self, source_path: S, destination_path: Option<D>)
where S: AsRef<Path>,
      D: AsRef<Path>
```

Reply:
That still failed, but I managed to work around the inference issue by forcing `D` to be `PathBuf` in the code generated by the proc macro for the `None` case.